src/Symfony/Component/Serializer/Mapping/AttributeMetadata.php

```diff
@@ -263,6 +317,6 @@ public function merge(AttributeMetadataInterface $attributeMetadata)
      */
     public function __sleep(): array
     {
-        return ['name', 'groups', 'maxDepth', 'serializedName', 'ignore', 'normalizationContexts', 'denormalizationContexts'];
+        return ['name', 'groups', 'maxDepth', 'serializedNames', 'ignore', 'normalizationContexts', 'denormalizationContexts'];
```

Comment:
The change in the name will break existing serialized payloads. We should try to preserve compatibility instead. I feel like this should be done by preserving the name and allowing+handling `string|array|null` for the property.

Reply:
Tu be sure, we keep `serializedName` property and not change new public methods name ? (`getSerializedNames()`)

WDYT about always transform `serializedName` to array at unserialized ?

```php
public function __wakeup()
{
    if (null === $this->serializedName) {
        $this->serializedName = [];
    } elseif (\is_string($this->serializedName)) {
        $this->serializedName = [
            '*' => $this->serializedName,
        ];
    }
}
```

Reply:
> Tu be sure, we keep serializedName property and not change new public methods name ? (getSerializedNames())

yes

> WDYT about always transform serializedName to array at unserialize ?

works for me if that makes the rest simpler